### ISSUE TITLE
[Question] Amd Graphics Card

### ISSUE THREAD

**Issue by FeyrisTan** [the-database/mpv-upscale-2x_animejanai#24]
Will it be possible to do this (in realtime) with an amd card in the near future?
I sold my 1070ti and I'm going to get a 6800xt + 13600k. Was also considering the rtx 4070 but 12gb vram and 192 bit made me look at amd.

**Comment by 422415** [comment]
We utilize a specific NVidia function called a TensorRT Engine, which gives us the speed needed for realtime of Compact models. It is technically possible in realtime with the smallest model(SuperUltraCompact) with AMD however they are generations behind in speed for this sort of task.

**Comment by hooke007** [comment]
9980hk+pro 5600m
720p x2 → 1440p
janai super ultra compact works fine
<img width=500 src="https://github.com/the-database/mpv-upscale-2x_animejanai/assets/41094733/f3890b0d-7c90-4b33-adfb-772e78632d02">

**Comment by animeojisan** [comment]
Is it possible to integrate some of the features of mpv-lazy-2023V5 into the mpv-upscale-2x_animejanai player? I think it would work that way.

**Comment by hooke007** [comment]
Both are mpv players. No barriers if you know how to use mpv.

**Comment by animeojisan** [comment]
What are the minimum required files and folders to copy from mpv-lazy-2023V5 to mpv-upscale-2x_animejanai?

**Comment by animeojisan** [comment]
For example, will 2x_AnimeJaNai_V2_SuperUltraCompact_100k.onnx also work? I have seen some information that it only works with FP32. I don't know if the above ONNX is FP32 or not.

**Comment by hooke007** [comment]
No.
But fp32 janai models are bundled in vsLite.

**Comment by hooke007** [comment]
OFF topic:
Though fp16 onnx is almost meaningless for nvidia TRT. trt beackend doesn’t use onnx directly, fp16 engine can be converted at runtime, which is actually used by this backend.

**Comment by animeojisan** [comment]
Thanks.
Can I assume that animejanaiV2L1.onnx and 2x_AnimeJaNai_V2_SuperUltraCompact_100k.onnx are identical? Is the image quality the same?

**Comment by hooke007** [comment]
In our use case, I don’t see any obvious diff.

**Comment by animeojisan** [comment]
Very helpful! Thank you.

**Comment by bananakid** [comment]
> 9980hk+pro 5600m 720p x2 → 1440p janai super ultra compact works fine <img alt="" width="500" src="https://user-images.githubusercontent.com/41094733/261125389-f3890b0d-7c90-4b33-adfb-772e78632d02.jpg">
> No. But fp32 janai models are bundled in vsLite.
> not sure. For generic gpu, mpv.exe + vsLite can allow you to use janai.

Hey @hooke007, would you kindly clarify how to reference Janai SuperUltraCompact (that are bundled in vsLite) in `conf` file to actually use it for video playback? I'm running the rare **AMD Radeon Pro 5600M** with Windows 10 too. I've extracted vsLite to `mvp-lazy` location and I'm familiar with loading GLSL shaders and customizing `conf`. I just need a little help on where and what files should be loaded. BTW let me know if you're using your GPU in MBP or not.

P.S. I have tried injecting `shaders` and `profiles` from `mpv-upscale-2x_animejanai` to `mvp-lazy+vsLib` and specifying this in `profiles.conf`:
```
[default]
profile=upscale-on-ultracompact4x
```
And while `Stats > Filters` shows `vapoursynth [buffered-frames=1 concurrent-frames=99 file=~~/shaders/animejanai_v2_ultracompact.vpy]` this is the result:
![mvp-lazy](https://github.com/the-database/mpv-upscale-2x_animejanai/assets/17095595/f04cd318-0944-4f5b-9c91-fccac0f3bfb9)

**Comment by hooke007** [comment]
It isn't a shader. I never tried to mix this project with mine.

**Comment by bananakid** [comment]
@hooke007, thanks so much for taking time to respond, I succeeded with mpv-lazy+vsLite (via Shift+5). Probably `SR_ESRGAN_DML.vpy` works good for low-quality source file. However I wasn't impressed when I compared it to `fsrcnnx_x2_8-0-4-1_lineart.glsl`+`ssimsuperres.glsl` shader combo with a 720p of a decent bitrate (1280×720, ~3500 Kbps), shaders looks nicer and runs faster, [here's a comparison](https://slow.pics/c/Ci7QfMy1) (Kobayashi-san Chi No Maid Dragon S1E1).

**Comment by bananakid** [comment]
@hooke007, thanks so much for taking time to respond and for implementing the support in `mpv-lazy`, I succeeded with `mpv-lazy+vsLite` (via `Shift+5`). The quality is incredible and definitely outperforms my typical fsrcnnx_x2_8-0-4-1_lineart.glsl`+`ssimsuperres.glsl shader combo with an anime file of 720p ~3500 Kbps upscaled to 1440p. However the performance has a lot to be desired, frames are dropping and you can't really watch the result. However GPU utilization is around 80%. Is there something else I can tweak for it to run better with **AMD Radeon Pro 5600M**?

**Comment by hooke007** [comment]
I have no idea what happened on your side. I did successfully enjoyed the full episode of the anime (no dropping frames) in the last test. And now I am using the new apple silicon macbook, so I cannot help more.

**Comment by bananakid** [comment]
@hooke007 thank you for taking time to respond and for your articles in Discussions of mpv-lazy. I figured out ONNX models are included in vs-mlt [contrib-models.v14.test.7z](https://github.com/AmusementClub/vs-mlrt/releases). I will try older GPU driver and older mvp version, as well as different test file. I'll post an update in case I find a solution.

**Comment by hooke007** [comment]
Emm...The models you want are also included in vsLite.

**Comment by hooke007** [comment]
Emm...The models you want are also included in vsLite.

**Comment by bananakid** [comment]
@hooke007 yeah, I just linked model source in case model ever gets updated.

I downloaded a couple of other files and they both play OK:
```
2. h264 (H.264/AVC/MPEG-4 AVC/MPEG-4 part 10), 8-bit, 960x720, ~3450 Kbps, 23.976 fps (0 dropped frames for 6 min 30 sec)
3. h264 (H.264/AVC/MPEG-4 AVC/MPEG-4 part 10), 10-bit (Hi10P), 966x720, ~6000 Kbps, 23.976 fps (30 dropped frames for 6 min 30 sec)
```
It looks like the test file №1 I used is encoded with some issue. According to report it's encoded exactly the same as file №2, even bitrate is about 3500 Kbps, except one detail: files №2-3 (above, play OK) report in mpv Statistics `Video: h264 (H.264/AVC/MPEG-4 AVC/MPEG-4 part 10) (hwdec: d3d11va)` and file №1 (unplayable) reports `Video: h264 (H.264/AVC/MPEG-4 AVC/MPEG-4 part 10)`. So the problematic file is missing `(hwdec: d3d11va)`. The only other difference is that first file uses OGG Vorbis as audio codec and other two use AAC. The console output is exactly the same.

I have also obtained a different file from the same source and it has the same problem too (no `(hwdec: d3d11va)` in report and dropping frames constantly). Here's my mpv.conf:
```
vo = gpu-next
gpu-context = d3d11
hwdec = d3d11va
hwdec-codecs = h264,hevc,vp8,vp9,av1,prores
vd-lavc-dr=no
```

Do you happen to have any clue what may be causing this? I tried switching to gpu (from gpu-next), vulkan (for gpu-context and hwdec), it didn't help.

**Comment by hooke007** [comment]
> So the problematic file is missing (hwdec: d3d11va)

No issue here. Hi10p could not be decoded by most GPU.

**Comment by bananakid** [comment]
After some further testing I came to conclusion there's no problem for AMD Radeon Pro 5600M to use `animejanaiV2L1.onnx` via `SR_ESRGAN_DML.vpy` except the file resolution limitation: `960×720` upscales and plays flawlessly at ±60fps (h264, h265, 8-bit, 10-bit, any bitrate, any audio codec) while `1280×720` gets constant frame dropping (any codec, any color depth bits, any bitrate, any audio codec).

So thanks again @hooke007 for continuous support, you stated correctly in the very beginning regarding the weakness of GPU. But I'm still happy to know I can enjoy the next to worst quality being properly upscaled. After all newish releases can be downloaded in 1080p/2160p, but some 4:3 classic is often in 720p (`960×720`) at most.

I hope my observations will be a worthy contribution to data on AMD Radeon Pro 5600M capabilities.

**Comment by hooke007** [comment]
For DML, `lt_hd` only set the max res of 1280x720. It would terminate the process if the res of clip over it.

**Comment by 422415** [comment]
AMD cards are now supported with release of 3.4.0
